e2e/cypress/integration/messaging/pinned_posts_part2_spec.js

```diff
@@ -0,0 +1,257 @@
+        cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+        cy.postMessage('Hello');
+
+        cy.getLastPostId().then((postId) => {
+            // # Pin the post
+            pinPost(postId);
+
+            // # Now open RHS
+            cy.get(`#post_${postId}`).trigger('mouseover');
+            cy.get(`#CENTER_commentIcon_${postId}`).click({force: true});
+
+            // * Verify that pinned icon is there both in center and RHS
+            cy.get(`#rhsPost_${postId} .post-pre-header`).should('have.text', 'Pinned');
+            cy.get(`#post_${postId} .post-pre-header`).should('have.text', 'Pinned');
+
+            // # Un-pin the post from RHS
+            unpinPost(postId);
+
+            // * Verify that pinned icon is gone from center and RHS
+            cy.get(`#rhsPost_${postId} .post-pre-header`).should('not.have.text', 'Pinned');
+            cy.get(`#post_${postId} .post-pre-header`).should('not.have.text', 'Pinned');
+
+            // # Click pin icon
+            cy.get('#channelHeaderPinButton').should('be.visible').click();
+
+            // * Should not have any pinned posts
+            cy.get('#search-items-container .no-results__title').should('have.text', 'No pinned posts yet');
+        });
+    });
+
+    it('MM-T2174 Switching channels in center also changes pinned posts RHS', () => {
+        // # Login
+        cy.apiLogin(testUser);
+
+        // # Visit a test channel
+        cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+
+        cy.getLastPostId().then((postId) => {
+            // # Pin the post
+            pinPost(postId);
+
+            // # Click pin icon
+            cy.get('#channelHeaderPinButton').should('be.visible').click();
+
+            // * Number of pinned posts in RHS should be 1
+            cy.findByTestId('search-item-container').should('have.length', 1);
+
+            // # Go to town square
+            cy.get('#sidebarItem_town-square').should('be.visible').click();
+
+            // * Should not have any pinned posts
+            cy.get('#search-items-container .no-results__title').should('have.text', 'No pinned posts yet');
+
+            // * Number of pinned posts in RHS should be 0
+            cy.findByTestId('search-item-container').should('not.exist');
+
+            // # Visit a test channel
+            cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+
+            // # Unpin the post again
+            unpinPost(postId);
+        });
+    });
+
+    it('MM-T2175 Pin a post in a DM channel Pin a post while viewing empty pinned post RHS', () => {
+        // # Create DM Channel
+        cy.apiCreateDirectChannel([testUser.id, otherUser.id]).then(() => {
+            cy.apiLogin(testUser);
+
+            // # Visit the DM channel
+            cy.visit(`/${testTeam.name}/channels/${testUser.id}__${otherUser.id}`);
+
+            // # Click pin icon
+            cy.get('#channelHeaderPinButton').should('be.visible').click();
```

Comment:
Failing intermittently on me. After visiting a page via URL, good practice is to add timeout when getting the first element.
```javascript
cy.get('#channelHeaderPinButton', {timeout: TIMEOUTS.ONE_MIN}).should('be.visible').click();
```